### PR TITLE
[SQL] Propagate monotonicity information through flatmap operators

### LINKED ISSUE #2529: LATENESS doesn't propagate through unnest
In this program the `flattened` view  does not inherit the lateness annotation from `my_table`. The aggregates help illustrate the issue: the input to `agg1` gets GC'd, but the input to `agg2` doesn't. 

```sql
CREATE TABLE my_table(
  id bigint,
  ts timestamp NOT NULL LATENESS INTERVAL 1 days,
  data INT ARRAY
);

CREATE VIEW agg1 as 
SELECT
    MAX(id)
FROM
    my_table
GROUP BY ts;

CREATE VIEW flattened AS
SELECT
  id,
  ts,
  v
FROM my_table, unnest(data) as v;

CREATE VIEW agg2 as 
SELECT
    MAX(id)
FROM
    flattened
GROUP BY ts;
```

### PR DESCRIPTION
**Pull request by mihaibudiu**
Fixes #2529

### CODE REVIEW

**Review by ryzhyk — APPROVED** [review]
I tested this on a bigger example and it works correctly.

File: sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/streaming/StreamingTests.java [hunk 1]
```diff
@@ -1645,6 +1643,44 @@ date_time TIMESTAMP(3) NOT NULL LATENESS INTERVAL 4 SECONDS
...
+            int count = 0;
+
+            @Override
+            public void postorder(DBSPIntegrateTraceRetainKeysOperator operator) {
+                this.count++;
+            }
+
+            @Override
+            public void endVisit() {
+                Assert.assertEquals(4, this.count);
```
Comment (ryzhyk) [inline sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/streaming/StreamingTests.java:1678]: Do we want to also test that the circuit computes correct results?

Comment (mihaibudiu) [reply sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/streaming/StreamingTests.java:1678]: On which input?

Comment (ryzhyk) [reply sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/streaming/StreamingTests.java:1678]: Any minimal test input.

Comment (mihaibudiu) [reply sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/streaming/StreamingTests.java:1678]: I will add an I/O test in the next PR

**Pull request merged**